plugin/src/main/kotlin/com/automattic/android/BuildEnvironment.kt

```diff
@@ -1,19 +1,37 @@
 package com.automattic.android.publish
 
+const val BRANCH_NAME_ARGUMENT_NAME = "branch-name"
+const val SHA1_ARGUMENT_NAME = "sha1"
+
+private val requiredArgumentsErrorMessage: String by lazy {
+    "--$BRANCH_NAME_ARGUMENT_NAME={branch-name} --$SHA1_ARGUMENT_NAME={sha1-commit-hash}" +
+        " command line arguments are required"
+}
+
+private const val DONT_PUBLISH_ERROR_MESSAGE = "Since this build is not from a tag and it's on a " +
+    "branch without a pull request url, it shouldn't be published to S3."
+
 data class BuildEnvironment(
```

Comment:
I am not sure if this class should throw exceptions like this. I might need to re-work this once I understand all the usages of it: probably `calculateVersionName`, `publishToS3` & `shouldPublishToS3` tasks.